csrc/transform_replay.cpp

```diff
@@ -170,7 +170,10 @@ TensorDomain* TransformReplay::fullSelfReplay(
 
   NVF_ERROR(
       new_self_root->maybeRoot().size() == self->maybeRoot().size(),
-      "Invalid number of IterDomains provided.");
+      "Invalid number of IterDomains provided: ",
+      new_self_root->maybeRoot().size(),
+      " vs ",
+      self->maybeRoot().size());
```

Comment:
```suggestion
      new_self_root->maybeRoot(),
      " vs ",
      self->maybeRoot());
```
makes error messages more useful. Root domains are typically short so readers shouldn't have a problem counting.